include/swift/Driver/ToolChain.h

```diff
@@ -204,6 +242,10 @@ class ToolChain {
                                          StringRef sanitizer,
                                          bool shared=true) const;
 
+  /// For an index data job,
+  /// only make the index-file-path ones primary.
```

Comment:
Thinking about this more, I'm pretty sure this function is the wrong answer anyway. Whatever's doing indexing should just construct jobs appropriately so that the one input is marked as primary. But you don't have to change that in this PR.

Reply:
There's this test, driver-index.swift:
`// RUN: %swiftc_driver -driver-print-jobs -index-file -index-file-path %s %S/Inputs/SwiftModuleA.swift %s -o %t.output_for_index -index-store-path %t.index_store -module-name driver_index 2>&1 | %FileCheck %s`.
It names %s *twice* in the command line, and expects the frontend to see one %s, as the first input, as a primary input.
I introduced this function to preserve this behavior. It would be great if we could change the test!

Reply:
New round of comments where I've given up on improving `-index-file`: still, this is an implementation comment rather than an interface comment.